Sofa/Component/MechanicalLoad/src/sofa/component/mechanicalload/NodalLinearDampingForceField.inl

```diff
@@ -0,0 +1,349 @@
+* along with this program. If not, see <http://www.gnu.org/licenses/>.        *
+*******************************************************************************
+* Authors: The SOFA Team and external contributors (see Authors.txt)          *
+*                                                                             *
+* Contact information: contact@sofa-framework.org                             *
+******************************************************************************/
+#pragma once
+
+#include <sofa/component/mechanicalload/NodalLinearDampingForceField.h>
+
+#include <sofa/core/MechanicalParams.h>
+#include <sofa/core/behavior/BaseLocalForceFieldMatrix.h>
+
+namespace sofa::component::mechanicalload
+{
+
+
+template<class DataTypes>
+NodalLinearDampingForceField<DataTypes>::NodalLinearDampingForceField()
+    : d_dampingCoefficientVector(initData(&d_dampingCoefficientVector, "dampingCoefficientVector", "Vector of velocity damping coefficients (by cinematic dof and by node)"))
+    , d_dampingCoefficient(initData(&d_dampingCoefficient, "dampingCoefficient", "Node-constant and isotropic damping coefficient"))
+{
+    sofa::core::objectmodel::Base::addUpdateCallback("updateFromDampingCoefficientVector", {&d_dampingCoefficientVector}, [this](const core::DataTracker& )
+    {
+        msg_info() << "call back update: from dampingCoefficientVector";
+        return updateFromDampingCoefficientVector();
+    }, {});
+
+    sofa::core::objectmodel::Base::addUpdateCallback("updateFromSingleDampingCoefficient", {&d_dampingCoefficient}, [this](const core::DataTracker& )
+    {
+        msg_info() << "call back update: from dampingCoefficient";
+        return updateFromSingleDampingCoefficient();
+    }, {});
+}
+
+
+template<class DataTypes>
+sofa::core::objectmodel::ComponentState NodalLinearDampingForceField<DataTypes>::updateFromDampingCoefficientVector()
+{
+    // Check if the dampingCoefficientVector has a null size
+    VecDeriv dampingCoefficients = d_dampingCoefficientVector.getValue();
+    const unsigned int sizeCoefs = dampingCoefficients.size();
+    if(sizeCoefs == 0)
+    {
+        msg_error() << "Size of the \'dampingCoefficientVector\' vector is null";
+        return sofa::core::objectmodel::ComponentState::Invalid;
+    }
+
+    // Recover mstate size
+    const unsigned int sizeMState = this->mstate->getSize();
+
+
+    // If the sizes of the dampingCoefficientVector and the mechanical state mismatch
+    if(sizeCoefs != sizeMState && sizeCoefs != 1)
+    {
+        msg_error() << "Size of the \'dampingCoefficientVector\' vector does not fit the associated MechanicalObject size";
+        return sofa::core::objectmodel::ComponentState::Invalid;
+    }
+
+
+    // If size=1, make a constant vector which size fits the MechanicalObject
+    if(sizeCoefs == 1)
+    {
+        sofa::helper::WriteAccessor<Data<VecDeriv> > accessorCoefficient = this->d_dampingCoefficientVector;
```

Comment:
in this case, you could also fallback to the single-value behavior. You set `d_dampingCoefficient` and then call `updateFromSingleDampingCoefficient`. No?